x-pack/plugin/profiling/src/main/java/org/elasticsearch/xpack/profiling/FrameGroupID.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.profiling;
+
+import java.util.Objects;
+
+public class FrameGroupID {
```

Comment:
As this is a utility class: Can we declare it `final` and implement a private default constructor? This avoids that anybody creates instances of this class by mistake.

Reply:
Nice, thanks!